src/lsstseries/analysis/structurefunction2.py

```diff
@@ -69,6 +70,13 @@ def calc_sf2(
 
             # Mask on band
             times = np.array(time)[band_mask]
+
+            # if times are all `None`, we assume equidistant times between
+            # measurements. To do so, we'll create an array of integers
+            # from 0 to N where N is the number of flux values for this band.
+            if np.all(np.equal(times, None)):
```

Comment:
Small comment and it is fine to leave it as-is: Alternatively (or in addition) you could check if `times` itself is None. This means that the user wouldn't need to allocate an array of Nones.

Reply:
I was a little on the fence about supporting that use case, based only the existing docstring. But I do think it's a good idea. So I'll work in the changes necessary to support that.